File: FlowCrypt/src/main/java/com/flowcrypt/email/js/PgpKeyInfo.java
/*
 * © 2016-2018 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.js;

import android.os.Parcel;
import android.os.Parcelable;

public class PgpKeyInfo implements Parcelable {

  public static final Creator<PgpKeyInfo> CREATOR = new Creator<PgpKeyInfo>() {
    @Override
    public PgpKeyInfo createFromParcel(Parcel source) {
      return new PgpKeyInfo(source);
    }

    @Override
    public PgpKeyInfo[] newArray(int size) {
      return new PgpKeyInfo[size];
    }
  };

  private final String longid;
  private final String keyValue;

  public PgpKeyInfo(String keyValue, String longid) {
    this.keyValue = keyValue;
    this.longid = longid;
  }

  protected PgpKeyInfo(Parcel in) {
    this.longid = in.readString();
    this.keyValue = in.readString();
  }

  @Override
  public int describeContents() {
    return 0;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeString(this.longid);
    dest.writeString(this.keyValue);
  }

  public String getLongid() {
    return longid;
  }

  public String getPrivate() {
    return keyValue;
  }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/model/PgpKeyInfo.kt
/*
 * © 2016-2019 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.model

import android.os.Parcel
import android.os.Parcelable

data class PgpKeyInfo constructor(val longid: String,
                                  val private: String?,
                                  val pubKey: String) : Parcelable {
  constructor(source: Parcel) : this(
      source.readString()!!,
      source.readString(),
      source.readString()!!
  )

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) =
      with(dest) {
        writeString(longid)
        writeString(private)
        writeString(pubKey)
      }

  companion object {
    @JvmField
    val CREATOR: Parcelable.Creator<PgpKeyInfo> = object : Parcelable.Creator<PgpKeyInfo> {
      override fun createFromParcel(source: Parcel): PgpKeyInfo = PgpKeyInfo(source)
      override fun newArray(size: Int): Array<PgpKeyInfo?> = arrayOfNulls(size)
    }
  }
}
